### DISCUSSION TITLE
OAuth with token Google

### DISCUSSION THREAD

**Discussion by Phantom-KNA** [supabase/supabase#16873] [Questions]
I have implemented authentication with Google in Android, I am using `SignInWithIdToken` passing Google's response IdToken or RequestIdToken as a parameter but it gives an error, that I am missing.

`{"error":"invalid request","error_description":"Bad ID token"}`

**Comment by AwesomeGameConcepts** [comment]
I have the same issue with Unity and GooglePlay SDK. Did you found a workaround ?

`
public void PerformSignIn()
    {
#if UNITY_ANDROID
        if (_signInStatus == SignInStatus.Success)
        {
            PlayGamesPlatform.Instance.RequestServerSideAccess(true, SignInSupabase);
        }
        else
        {
            PlayGamesPlatform.Instance.ManuallyAuthenticate(ProcessAuthentication);
        }
#endif
    }
    private async void SignInSupabase(string idToken)
    {
#if UNITY_ANDROID
        try
        {
            Supabase.Gotrue.Session lSession = await _supabase.AuthClient.SignInWithIdToken(Constants.Provider.Google, idToken);
            OnLogIn();
        }
        catch(Exception lEx)
        {
            await MessageBoxManager.Instance.DisplayMesageAndWaitClose("fail", lEx.Message);
        }
#endif
    }
`
